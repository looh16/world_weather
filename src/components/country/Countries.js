import PropTypes from 'prop-types';
import * as BsIcon from 'react-icons/bs';
import { useNavigate } from 'react-router-dom';
import Image from 'react-bootstrap/Image';

const Countries = ({ country }) => {
  const navigate = useNavigate();
  const navigateDetails = () => {
    navigate('/details', { state: { lat: country.latlng[0], lon: country.latlng[1], countryName: country.name.official } });
  };

  return (
    <div className="container d-flex flex-column justify-content-center">

      <BsIcon.BsFillArrowRightCircleFill className="align-self-end mt-2" name={country.name.official} onClick={navigateDetails} key={country.name.official} />
      <div>
        <h4>{country.name.official}</h4>
        <h5>
          Continent: &nbsp;
          {country.continents[0]}
        </h5>
        <p>
          <Image
            fluid
            src={country.flags.png}
            rounded
          />
        </p>
      </div>
    </div>
  );
};

Countries.propTypes = {
  name: PropTypes.string,
}.isRequired;

export default Countries;
