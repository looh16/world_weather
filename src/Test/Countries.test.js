import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Countries from '../components/country/Countries';

test('renders correctly', () => {
  const country = {
    name : {
      official : "Mozambique"
    },
    continents: ["Africa","Asia"],
    "flags": {
      "png": "https://flagcdn.com/w320/fi.png",
    },
  }
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Countries country={country} />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
