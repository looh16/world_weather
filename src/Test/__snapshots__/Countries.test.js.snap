// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container d-flex flex-column justify-content-center"
>
  <svg
    className="align-self-end mt-2"
    fill="currentColor"
    height="1em"
    name="Mozambique"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
    />
  </svg>
  <div>
    <h4>
      Mozambique
    </h4>
    <h5>
      Continent: Â 
      Africa
    </h5>
    <p>
      <img
        className="img-fluid rounded"
        src="https://flagcdn.com/w320/fi.png"
      />
    </p>
  </div>
</div>
`;
